README.md

```diff
@@ -269,6 +269,16 @@ A `JsonApiClient::ResultSet` object should be paginatable with both `kaminari` a
 Person.where(name: 'Jeff').all
 ```
 
+## Headers
+
+You can specify the headers to send with all requests within a block with `with_headers`.
```

Comment:
I don't know that we have to patch this gem, but we do need to resolve the multi-threaded use-case before we use this gem in any threaded code. We can use semaphores on the client end around any call to this gem and serialize requests that way. Pretty heavy handed, but if it's what we have to do. Personally since we've forked this gem, I'd just fix it. That said, we might want to dig a little further and find out if there's anything else that's storing state on the class that would be vulnerable. Might be that this gem has more than just this method having thread safety issues.